### PR TITLE
Updating tools/picard tool to 2.26.3 2.26.4 3.1.1

### PR DESCRIPTION
**Pull request by planemo-autoupdate**
Hello! This is an automated update of the following tool: **tools/picard**. I created this PR because I think the tool's main dependency is out of date, i.e. there is a newer version available through conda.

Please see https://github.com/planemo-autoupdate/autoupdate for more information.

### CODE REVIEW

**Comment by bwlang** [comment]
It seems that the logs of problems preventing tests passing are no longer cached... @bernt-matthias  do you what the problem is?
We are seeing some libraries that are double counting some bases and yielding > 100% when summing the various categories.  
I think [this change](https://github.com/broadinstitute/picard/commit/a746ec0ae5e48b1e59646963067a5fc287224e35) in more recent versions of picard should resolve the problem, but it's not available yet in galaxy
Let me know how I can help

**Comment by bgruening** [comment]
@bwlang you can help reviewing this PR now :)

**Comment by bgruening** [comment]
Ok, this seems to be green. All tests passing. The one linting error is from shared macros that I don't want to touch in this PR.

The last commit adds some reformatting, you can avoid this commit I hope. The reformatting is not strictly needed atm, but picard is changing its CLI interface so at some point we need to adopt ... lets see if I find motivation for the other tools.

Feel free to merge as it is.

**Review by bwlang — COMMENTED** [review]
Overall this looks good to me.
I found only a couple of inconsistencies that might be worth addressing.

for later, here is a list tools that still needing conversion to -ARG instead of ARG= style.
tools/picard/picard_AddOrReplaceReadGroups.xml
tools/picard/picard_CleanSam.xml
tools/picard/picard_EstimateLibraryComplexity.xml
tools/picard/picard_FastqToSam.xml
tools/picard/picard_FilterSamReads.xml
tools/picard/picard_MarkDuplicates.xml
tools/picard/picard_MarkDuplicatesWithMateCigar.xml
tools/picard/picard_MeanQualityByCycle.xml
tools/picard/picard_MergeBamAlignment.xml
tools/picard/picard_MergeSamFiles.xml
tools/picard/picard_QualityScoreDistribution.xml
tools/picard/picard_ReorderSam.xml
tools/picard/picard_ReplaceSamHeader.xml
tools/picard/picard_RevertOriginalBaseQualitiesAndAddMateCigar.xml
tools/picard/picard_RevertSam.xml
tools/picard/picard_SortSam.xml
tools/picard/picard_ValidateSamFile.xml
this little sed might help (untested):
```
# Use sed to replace ARGUMENT='${value}' or ARGUMENT='value' with -ARGUMENT ${value} or -ARGUMENT value
sed -i -r "s/([A-Z_]+)='\$\{([^}]+)\}'/-\1 \$\{\2\}/g; s/([A-Z_]+)='([^']+)'/-\1 \2/g" "$filename"
```

**Review by bwlang — APPROVED** [review]

File: tools/picard/picard_BedToIntervalList.xml [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+    </macros>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
     @java_options@
     @symlink_element_identifier@
     #set $picard_dict = "localref.dict"
     #set $ref_fasta = "localref.fa"     ## This is done because picards "likes" .fa extension
 
-    ln -s "${reference_source.ref_file}" "${ref_fasta}" &&
+    ln -sf '${reference_source.ref_file}' '${ref_fasta}' &&
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/picard/picard_BedToIntervalList.xml:14]: Yes, but when a tool is restarted we need this maybe.

File: tools/picard/picard_BedToIntervalList.xml [hunk 2]
```diff
@@ -30,54 +32,48 @@
...
-      OUTPUT="${outFile}"
+      -INPUT '$escaped_element_identifier'
+      -OUTPUT '${outFile}'
 
-      SEQUENCE_DICTIONARY="${picard_dict}"
-      QUIET=true
-      VERBOSITY=ERROR
+      -SEQUENCE_DICTIONARY '${picard_dict}'
+      -QUIET true
+      -VERBOSITY ERROR
```
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/picard/picard_BedToIntervalList.xml:40]: I think I covered that in my latest commit. Thanks for checking.

**Comment by bgruening** [comment]
ok, all green again.

@bwlang can you maybe have a again a look. Especially at the generated commandline of the AddOrReplaceReadGroups tool? You can find all test results, with CLI, at https://github.com/galaxyproject/tools-iuc/actions/runs/8125263167?pr=3828 scroll down.

Thanks!

@bernt-matthias ready when you think its ready.

**Review by bwlang — APPROVED** [review]
I don't see any problems 
more consistent use of ln -sf, now all --
looking good.

**Review by bernt-matthias — APPROVED** [review]

**Pull request merged**